Project_Mira_DaU/Source/Project_Mira_DaU/Public/Characters/PlayerCharacter.cpp

```diff
@@ -21,6 +23,9 @@ void APlayerCharacter::BeginPlay()
     ObjectCapsule->OnComponentEndOverlap.AddDynamic(this, &APlayerCharacter::EndOverlap);
 
     GetCharacterMovement()->MaxStepHeight = MaxStepHeightPlayer;
+
+    AActor* FoundActor = UGameplayStatics::GetActorOfClass(GetWorld(), AGameManager::StaticClass());
+    GameManager = Cast<AGameManager>(FoundActor);
```

Comment:
Je te conseille d'ajouter un assert pour vérifier que GameManager est valide : 
check(GameManager != nullptr);

https://dev.epicgames.com/documentation/en-us/unreal-engine/asserts-in-unreal-engine

La validité de l'object GameManager n'est pas vérifiée quand il est utilisé plus tard dans BeginOverlap, du coup s'il n'est pas présent dans le world, ça provoquera un crash. Avec l'assert, tu t'assures que tes hypothèses sont justes, si ce n'est pas le cas, l'assert déclenche un crash et te permet de détecter au plus tôt toute erreur.